### ISSUE TITLE
Second run cannot be stopped

### ISSUE THREAD

**Issue by rojanu** [spalger/grunt-run#18]
I have got the following configuration and second stop doesn't stop the server, it prints `>> server already stopped`, which is not true.

After some debugging, I found that `clearPid` is invoked by `proc.on('close'` from within `trackBackgroundProc`

grunt-run: 0.5.1
node: 0.10.40
```
run: {
server: {
    args: ['server.js'],
    options: {
      cwd: 'site',
      wait: false,
      quiet: true,
      failOnError: true
    }
  }
}
```

```
  grunt.registerTask('one', function(){grunt.log.ok('First run')});
  grunt.registerTask('two', function(){grunt.log.ok('Second run')});

  grunt.registerTask('myrun', ['env:csa','run:server',
                               'one',
                               'stop:server',
                               'run:server',
                               'two',
                               'stop:server']);
```

**Comment by rojanu** [comment]
After some more debugging, on close fired by the first process still shutting down and not having enough time.

Could we get a flag or something like that to make sure that 'stop:server' has finished before continuing?

**Comment by spalger** [comment]
That's a good idea in general I think, waiting for the process to close before ending the `stop:x` task

**Comment by yneves** [comment]
I proposed a fix here #26 but now that i see the idea of waiting for the process to close on `stop:x` task i like it better.
